crudApp/src/components/Home.js

```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useState } from "react";
+import { Button, Table } from "react-bootstrap";
+import { Link, useNavigate } from "react-router-dom";
+import { useSelector } from "react-redux";
+import { useDispatch } from "react-redux";
+import { deleteEmp } from "../actions/index";
+import { getEmp } from "../actions/index";
+
+import "bootstrap/dist/css/bootstrap.min.css";
+
+function Home() {
+  const dispatch = useDispatch();
+  const newState = useSelector((state) => state.changeTheState);
+  const history = useNavigate();
+  let [Employees, setEmployees] = useState([]);
+  useEffect(() => {
+    fetch("https://mocki.io/v1/ce596b10-8325-4673-b949-d3b23af5372b")
+      .then((response) => response.json())
+      .then((data) => {
+        setEmployees(data);
+      })
+      .then(() => console.log(Employees));
+  });
+
+  function handleGet() {
+    dispatch(getEmp(Employees));
+  }
+
+  function handleDelete(id) {
+    var index = Employees.map((e) => {
+      return e.id;
+    }).indexOf(id);
+    dispatch(deleteEmp(index));
+    history("/");
+  }
+  function handleEdit(id, age, name) {
+    localStorage.setItem("Name", name);
+    localStorage.setItem("Age", age);
+    localStorage.setItem("id", id);
+  }
+  return (
+    <>
+      <div style={{ margin: "10rem" }}>
+        <Table striped bordered hover size="sm">
+          <thead>
+            <tr>
+              <th>Name</th>
+              <th>Age</th>
+              <th>Actions</th>
+            </tr>
+          </thead>
+          <tbody>
+            {newState && newState.length > 0
+              ? newState.map((item) => {
```

Comment:
what do you think the diff between these two conditions are? do you really need both?
newState && newState.length ?

Reply:
Actually I was trying to make sure newState isn't null otherwise newState.length will throw an error. But I think since I am passing initialstate with an empty array, those two conditions are redundant.